### PR TITLE
Removes bad `nodamage` var from projectiles, fixes Juggernaut / Rust Walker projectiles doing zero damage

### LINKED ISSUE #73756: "Scare Tactics" gone wrong - Juggernaut "Gauntlet Echo" does nothing.
## Reproduction:
Be cult juggernaut.
Launch your Gauntlet Echo at a noncult human, both at a distance and at point blank.
Both clearly connect and even have a blood spray effect.
The human is 100% healthy. Not so much as a little bloodloss even.

image

I always thought this attack was kinda weak. Wonder how long it's been like this? So much for a "devastating" attack.

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Juggernaut and Rust Walker projectiles were subtyped off of magic, which is `nodamage`.

- The juggernaut actually had a copy+paste error with their type `on_hit` which caused none of their special effects on hit ("relative patching catches this")

- Then I realized projectiles have this var `nodamage` which is, for all intents and purposes, just `damage > 0`. it's not checked for pacifism, it's just that. This is dumb. So very dumb, so I removed it. 
    - There are, however, a few situations which used it in a unique way, such as the blast wave cannon. This is why I replaced it with a proc, `is_hostile_projectile`, for certain situations to actually find out if the projectile is damaging. Projectiles can override this on a per type basis by default, damaging projectiles = hostile. 
    - This has a chance to break some things, but I ... kinda doubt it will. 

Fixes #73756

## Why It's Good For The Game

Projectiles that act as they should, less dumb vars

## Changelog

:cl: Melbert
fix: Fixes Juggernaut / Rust Walker projectiles doing zero damage
fix: Fixes Juggernaut projectiles not doing bonus damage to nearby structures
code: Removed projectile nodamage var, replaces it with just checking for damage 
/:cl:

### CODE REVIEW

**Comment by VladinXXV** [comment]
>  ...basis by buy default, 

...basis but by default,

**Comment by timothymtorres** [comment]
Does this fix #49885

File: code/game/objects/items/robot/items/tools.dm [hunk 1]
```diff
@@ -138,12 +138,11 @@
 
 /obj/item/borg/projectile_dampen/proc/process_usage(delta_time)
 	var/usage = 0
-	for(var/to_track in tracked)
-		var/obj/projectile/projectil = to_track
-		if(!projectil.stun && projectil.nodamage) //No damage
+	for(var/obj/projectile/projectil as anything in tracked)
```
Comment (san7890) [inline code/game/objects/items/robot/items/tools.dm:141]: man this var name sucks

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/herald.dm [hunk 2]
```diff
@@ -226,21 +226,22 @@
...
-			nodamage = TRUE
+			damage = 0
+
 	. = ..()
 	if(ismineralturf(target))
 		var/turf/closed/mineral/rock_target = target
 		rock_target.gets_drilled()
 
+/obj/projectile/herald/teleshot
+	name ="golden bolt"
```
Comment (san7890) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/herald.dm:241]:
```suggestion
	name = "golden bolt"
```

**Review by san7890 — CHANGES_REQUESTED** [review]
nit because you're probably the first to touch this code since whenever these were first implemented or something

**Pull request merged**